packages/codecs-numbers/src/i16.ts

```diff
@@ -3,12 +3,12 @@ import { combineCodec, FixedSizeCodec, FixedSizeDecoder, FixedSizeEncoder } from
 import { NumberCodecConfig } from './common';
 import { numberDecoderFactory, numberEncoderFactory } from './utils';
 
-export const getI16Encoder = (config: NumberCodecConfig = {}): FixedSizeEncoder<number, 2> =>
+export const getI16Encoder = (config: NumberCodecConfig = {}): FixedSizeEncoder<bigint | number, 2> =>
     numberEncoderFactory({
         config,
         name: 'i16',
         range: [-Number('0x7fff') - 1, Number('0x7fff')],
-        set: (view, value, le) => view.setInt16(0, value, le),
+        set: (view, value, le) => view.setInt16(0, typeof value === 'bigint' ? Number(value) : value, le),
```

Comment:
nit: I suppose you could just use `Number(value)` here, right? Since I don't think there is an overhead to casting a number to a number.